src/lib/controllers/config-manifest/config-manifest.ts

```diff
@@ -0,0 +1,237 @@
+		}
+
+		const schemas = await Promise.all(
+			paths.map(
+				p =>
+					loadFromUri({
+						uri: productRepoURI,
+						path: p,
+						errorMessage:
+							'Unable to find config-manifest.yml. See documentation_link for more info.\n',
+					}) as Promise<ConfigManifestRaw>,
+			),
+		);
+		return new ConfigManifest(
+			schemas.map(schema => {
+				return { schema };
+			}),
+		);
+	}
+
+	/**
+	 * Applies when condition to JsonSchema
+	 * @param obj
+	 * @param {string} key
+	 */
+	private static applyWhenCondition(obj: any, key: string): void {
+		const properties = _.get(obj, key);
+		const conditions: any = {};
+		const anyOf: any = [];
+		_.forIn(properties, function(val, key) {
+			if (_.includes(_.get(val, 'when'), '==')) {
+				const depArray = _.split(_.get(val, 'when'), '==');
+				const dependency = _.trim(depArray[0]);
+				const value = _.trim(depArray[1], `'" `);
+				if (!_.get(conditions, dependency, false)) {
+					conditions[dependency] = { [value]: [key] };
+				} else if (_.get(conditions, [dependency, value])) {
+					conditions[dependency][value].push(key);
+				} else {
+					conditions[dependency][value] = [key];
+				}
+			}
+		});
+
+		for (const requirements of _.values(conditions)) {
+			for (const requirement of _.values(requirements)) {
+				const conditionProperties: any = {};
+				for (const name of requirement) {
+					conditionProperties[name] = properties[name];
+				}
+				anyOf.push({
+					properties: conditionProperties,
+					required: requirement,
+				});
+			}
+		}
+
+		if (anyOf.length) {
+			obj['anyOf'] = anyOf;
+		}
+	}
+
+	/**
+	 * Recursively transforms current config-manifest schema format to JSONSchema.
+	 * It will be replaced by JellySchema and ReConFix.
+	 * @param obj
+	 */
+	private static traverse(obj: any) {
+		// Convert array of properties to object
+		_.forIn(obj, function(val, key) {
+			if (_.isArray(val) && key === 'properties') {
+				obj['properties'] = _.reduce(
+					val,
+					(o, v) => {
+						return _.merge(o, v);
+					},
+					{},
+				);
+				ConfigManifest.applyWhenCondition(obj, key);
+				obj['additionalProperties'] = true;
+			}
+		});
+
+		// Add transformations of special syntax
+		_.forIn(obj, function(val, key) {
+			if (_.isObject(val) && key !== 'anyOf') {
+				// Add required as default, unless type ends with '?'
+				if (_.endsWith(_.get(val, 'type', ''), '?')) {
+					obj[key]['type'] = obj[key]['type'].slice(0, -1);
+				} else if (
+					!_.includes(['default', 'properties'], key) &&
+					!_.get(val, 'when', false)
+				) {
+					obj[key]['required'] = true;
+				}
+				// Handle 'hostname', 'email', 'uri' type transformations
+				if (
+					_.includes(
+						['hostname', 'email', 'uri', 'number'],
+						_.get(val, 'type', ''),
+					)
+				) {
+					obj[key]['format'] = obj[key]['type'];
+					obj[key]['type'] = 'string';
+				}
+
+				if (key === 'properties') {
+					for (const el of _.keys(val)) {
+						if (_.isObject(el)) {
```

Comment:
According to the lodash docs @Page- `_keys()` will:

> Creates an array of the own enumerable property names of object.
> Note: Non-object values are coerced to objects. See the ES spec for more details.

So this check can drop since it will always be an object...

Reply:
No, it coerces the input value to an object, the response will always be `number | string | Symbol` and never an "object" (I'm ignoring `Symbol` which probably counts as an object because I doubt that was the intention).

Reply:
In that case we can drop the check I think